ArcGISRuntimeSDKQt_CppSamples/Routing/NavigateARouteWithRerouting/NavigateARouteWithRerouting.cpp

```diff
@@ -0,0 +1,367 @@
+// [WriteFile Name=NavigateARouteWithRerouting, Category=Routing]
+// [Legal]
+// Copyright 2022 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifdef PCH_BUILD
+#include "pch.hpp"
+#endif // PCH_BUILD
+
+#include "NavigateARouteWithRerouting.h"
+
+#include "DirectionManeuverListModel.h"
+#include "GeometryEngine.h"
+#include "GraphicsOverlay.h"
+#include "Location.h"
+#include "LocationDisplay.h"
+#include "Map.h"
+#include "MapQuickView.h"
+#include "NavigationTypes.h"
+#include "Point.h"
+#include "ReroutingParameters.h"
+#include "Route.h"
+#include "RouteParameters.h"
+#include "RouteResult.h"
+#include "RouteTask.h"
+#include "RouteTracker.h"
+#include "RouteTrackerLocationDataSource.h"
+#include "SimpleMarkerSymbol.h"
+#include "SimulatedLocationDataSource.h"
+#include "SimulationParameters.h"
+#include "Stop.h"
+#include "TrackingDistance.h"
+#include "TrackingProgress.h"
+#include "TrackingStatus.h"
+#include "VoiceGuidance.h"
+
+#include <memory>
+#include <QDir>
+#include <QList>
+#include <QTextToSpeech>
+#include <QTime>
+#include <QUrl>
+
+
+using namespace Esri::ArcGISRuntime;
+
+namespace  {
+const Point conventionCenterPoint(-117.160386727, 32.706608, SpatialReference::wgs84());
+const Point aerospaceMuseumPoint(-117.147230, 32.730467, SpatialReference::wgs84());
```

Comment:
set these to the first and last points of the tourPolylineJson so the navigation properly ends
```suggestion
const Point conventionCenterPoint(-117.160386727066026, 32.706608204740171, SpatialReference::wgs84());
const Point aerospaceMuseumPoint(-117.146679999797271, 32.730351428946854, SpatialReference::wgs84());
```